js/compressed-token/src/actions/decompress.ts

```diff
@@ -0,0 +1,88 @@
+import {
+    ConfirmOptions,
+    PublicKey,
+    Signer,
+    TransactionSignature,
+} from '@solana/web3.js';
+import {
+    bn,
+    defaultTestStateTreeAccounts,
+    sendAndConfirmTx,
+    buildAndSignTx,
+    Rpc,
+} from '@lightprotocol/stateless.js';
+
+import { BN } from '@coral-xyz/anchor';
+
+import { dedupeSigner, getSigners } from './common';
+import { CompressedTokenProgram } from '../program';
+import { selectMinCompressedTokenAccountsForTransfer } from './transfer';
+
+/**
+ * Decompress compressed tokens
+ *
+ * @param rpc            Rpc to use
+ * @param payer          Payer of the transaction fees
+ * @param mint           Mint of the compressed token
+ * @param amount         Number of tokens to transfer
+ * @param owner          Owner of the compressed tokens
+ * @param toAddress      Destination **uncompressed** (associated) token account
+ *                       address.
+ * @param merkleTree     State tree account that any change compressed tokens should be
+ *                       inserted into. Defaults to a default state tree
+ *                       account.
+ * @param multiSigners   Signing accounts if `currentOwner` is a multisig
+ * @param confirmOptions Options for confirming the transaction
+ *
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function decompress(
+    rpc: Rpc,
+    payer: Signer,
+    mint: PublicKey,
+    amount: number | BN,
+    owner: Signer | PublicKey,
+    toAddress: PublicKey,
+    /// TODO: allow multiple
+    merkleTree: PublicKey = defaultTestStateTreeAccounts().merkleTree,
+    multiSigners: Signer[] = [],
+    confirmOptions?: ConfirmOptions,
+): Promise<TransactionSignature> {
+    const [currentOwnerPublicKey, signers] = getSigners(owner, multiSigners);
+
+    amount = bn(amount);
+
+    const compressedTokenAccounts = await rpc.getCompressedTokenAccountsByOwner(
+        currentOwnerPublicKey,
+        {
+            mint,
+        },
+    );
+
+    /// TODO: consider using a different selection algorithm
+    const [inputAccounts] = selectMinCompressedTokenAccountsForTransfer(
+        compressedTokenAccounts,
+        amount,
+    );
+
+    const proof = await rpc.getValidityProof(
+        inputAccounts.map(account => bn(account.compressedAccount.hash)),
+    );
+
+    const ixs = await CompressedTokenProgram.decompress({
+        payer: payer.publicKey,
+        inputCompressedTokenAccounts: inputAccounts,
+        toAddress: toAddress, // TODO: add check that it is a token account
+        amount: amount,
+        outputStateTree: merkleTree,
+        recentInputStateRootIndices: proof.rootIndices,
+        recentValidityProof: proof.compressedProof,
+    });
+
+    const { blockhash } = await rpc.getLatestBlockhash();
+    const additionalSigners = dedupeSigner(payer, signers);
+    const signedTx = buildAndSignTx(ixs, payer, blockhash, additionalSigners);
+    const txId = await sendAndConfirmTx(rpc, signedTx, confirmOptions);
+    return txId;
```

Comment:
on a general note, we could think about returning more information here such as which compressed accounts have been spent or created etc.

Reply:
yeah good idea, noted for a followup pr